maestro-cli/src/main/java/maestro/cli/report/HtmlTestSuiteReporter.kt

```diff
@@ -0,0 +1,127 @@
+package maestro.cli.report
+
+import maestro.cli.model.TestExecutionSummary
+import maestro.cli.report.TestDebugReporter
+import okio.Sink
+import okio.buffer
+import java.io.BufferedReader
+import java.io.FileReader
+import kotlinx.html.*
+import kotlinx.html.stream.appendHTML
+
+class HtmlTestSuiteReporter : TestSuiteReporter {
+  override fun report(summary: TestExecutionSummary, out: Sink) {
+    val bufferedOut = out.buffer()
+    val htmlContent = buildHtmlReport(summary)
+    bufferedOut.writeUtf8(htmlContent)
+    bufferedOut.close()
+  }
+
+  private fun getFailedTest(summary: TestExecutionSummary): Array<String>{
+    var failedTest = emptyArray<String>()
+    for (suite in summary.suites) {
+        for(flow in suite.flows){
+            if(flow.status.toString() == "ERROR"){
+                failedTest += flow.name
+            }
+        }
+    }
+    return failedTest
+  }
+
+  private fun buildHtmlReport(summary: TestExecutionSummary): String {
+    var failedTest = getFailedTest(summary)
+    return buildString {
+      appendHTML().html {
+        head {
+          title { +"Maestro Test Report" }
+          link(rel = "stylesheet", href = "https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css") {}
+        }
+        body {
+          summary.suites.forEach { suite ->
+            div(classes = "card mb-4") {
+              div(classes = "card-body") {
+                h1(classes = "mt-5 text-center") { +"Test Execution Summary" }
+                br{}
+                +"Test Result: ${if (suite.passed) "PASSED" else "FAILED"}"
+                br{}
+                +"Duration: ${suite.duration}"
+                br{}
+                br{}
+                div(classes = "card-group mb-4") {
+                  div(classes = "card") {
+                    div(classes = "card-body") {
+                      h5(classes = "card-title text-center") { +"Total Test" }
+                      h3(classes = "card-text text-center") { +"${suite.flows.size}" }
+                    }
+                  }
+                  div(classes = "card text-white bg-danger") {
+                    div(classes = "card-body") {
+                      h5(classes = "card-title text-center") { +"Failed Test" }
```

Comment:
```suggestion
                      h5(classes = "card-title text-center") { +"Failed Flows" }
```
```suggestion
                      h5(classes = "card-title text-center") { +"Failed Test" }
```

Reply:
This also needs to be pluralized correctly, eg it should say `Flow` when there is only one Flow and `Flows` when it's 0 or more than 1